### PR TITLE
During deployment, activate agent after applying PR patches

### PR DESCRIPTION
**Pull request by goughes**
#### Status
Tested

#### Description
If you have a PR that updates the agent base config (WMAgentConfig.py) and you try to apply it during agent deployment, it never updates. That is because the "activate-agent" function of the manage script copies the default config to /data/srv/wmagent/current/config before the PR patches are applied. 

https://github.com/dmwm/deployment/blob/2536c3673535d1e19515f3ca06c92db7007bf1a4/wmagent/manage#L102

This PR reverses the order so you can at least apply a WMAgentConfig.py patch during deployment by specifying the pull path to the file like this:

```
|diff --git a/etc/WMAgentConfig.py b/etc/WMAgentConfig.py
|index 6a0087de07..34049fcd79 100644
|--- a/etc/WMAgentConfig.py
|+++ b/etc/WMAgentConfig.py
--------------------------
File to patch: /data/srv/wmagent/current/sw/slc7_amd64_gcc630/cms/wmagent/1.2.6.patch1/etc/WMAgentConfig.py
patching file /data/srv/wmagent/current/sw/slc7_amd64_gcc630/cms/wmagent/1.2.6.patch1/etc/WMAgentConfig.py
```
Then you don't have to manually patch the agent config later (after `wmagent-mod-config` has already run), which is pretty annoying in some cases.

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
NA

#### External dependencies / deployment changes
NA

### CODE REVIEW

**Comment by amaltaro** [comment]
@todor-ivanov Todor, can you please get this PR tested while deploying the backfill agent on vocms0285?

**Comment by amaltaro** [comment]
I just tested this patch and it works for normal `src/python` patches. I didn't test for configuration like patches though. @goughes have you tested that use case? If so, then it's good to go from my side.

**Review by amaltaro — APPROVED** [review]

**Comment by goughes** [comment]
I tested this. The config patches won't apply automatically since they are out of the `src/python` directory, but you can specify the absolute path to patch the files. Like the example in the PR description.

```File to patch: /data/srv/wmagent/current/sw/slc7_amd64_gcc630/cms/wmagent/1.2.6.patch1/etc/WMAgentConfig.py```

**Comment by amaltaro** [comment]
Is this patch good to go from your side? If so let me know and remove the "do not merge" label please

**Comment by goughes** [comment]
It's good from my side.

**Comment by amaltaro** [comment]
Thanks

**Pull request merged**
